airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -120,3 +130,40 @@ def get_dataset_events(
     return dataset_event_collection_schema.dump(
         DatasetEventCollection(dataset_events=events, total_entries=total_entries)
     )
+
+
+@security.requires_access([(permissions.ACTION_CAN_CREATE, permissions.RESOURCE_DATASET)])
+@provide_session
+def post_dataset_event(session: Session = NEW_SESSION) -> APIResponse:
+    """Create an external dataset event. This endpoint is useful if you want to update a dataset and
+    trigger downstream DAG runs from external services.
+    """
+    try:
+        json_body = dataset_change_schema.load(get_json_request_dict())
+    except ValidationError as err:
+        raise BadRequest(detail=str(err))
+    uri = json_body["dataset_uri"]
+    external_source = request.remote_addr
+    user_id = getattr(current_user, "id", None)
```

Comment:
isn't the id how a typical FK->PK relationship is defined?
it seems appropriate to add user_id and make a FK to users table.  then one could always look up the username by joining?

Reply:
Yeah good point, this should probably just be a fk.

Reply:
true

Reply:
i think there's a mechanism to set to null when user deleted?

Reply:
Setting this to a (non-fk) username would be much more useful than having a null IMO.

Reply:
I don't stand in way here, don't really mind, but if you'll humor me I'll think it through out loud... 

Maybe i'm just hung up on the standard practice of using surrogate keys, normalization, etc but...

So, airflow provides a mechanism to deactivate users.  Interestingly, on the user edit form, it even says `don't delete users; it's a bad practice; just deactivate them`.

<img width="829" alt="image" src="https://github.com/apache/airflow/assets/15932138/bcff2dd8-0c8e-47c0-b4fe-de8d0493290c">

Additionally a username can be changed.  So I could take some action, change my username, and now you don't know that I took that action.  

Additionally, you could delete a user, have a new user added with same username, but it'd be a different "user".

I think your point that having a username is more useful than having a null is obviously true.  But I guess my thought is that, it's not a situation that should happen, because users should not be deleted (because of course by keeping them we don't run into these problems), and if an airflow cluster admin does that well, that's up to them.  But presuming they don't you have the benefits of referential integrity.

Interestingly the log table does not have a user_id column, which is a bit weird... probably should... but then there too i'd say it would make sense to record the db ID.  

Another option would be to _reject_ the deletion of a user when there are associated log or dataset events. That would seem reasonable too.

Reply:
I guess the other reason @uranusjr would be so that we can use standard ORM features such as relationships to get from user to dataset and vice versa but... i suppose you could say that you could still do so with custom join conditions 🤷

Reply:
brought this up on slack.  the suggestion is to not add user column at all right now since user handling will change with AIP-56.  and for auditing purposes you can add a record to the Log table.